test/integration/extensions/test_compression_plugin.py

```diff
@@ -0,0 +1,94 @@
+import os
+import textwrap
+
+from conan.test.assets.genconanfile import GenConanfile
+from conan.test.utils.tools import TestClient
+
+
+def test_compression_plugin_not_valid():
+    """Test an error is raised if the compression plugin is not valid"""
+
+    c = TestClient()
+    compression_plugin = textwrap.dedent(
+        """
+        def tar_compress(files, name, dest_dir, compresslevel=None, ref=None, recursive=False):
+            return None
+    """
+    )
+
+    c.save(
+        {
+            os.path.join(
+                c.cache_folder, "extensions", "plugins", "compression.py"
+            ): compression_plugin,
+            "conanfile.py": GenConanfile("pkg", "1.0"),
+        }
+    )
+    c.run("create .")
+    c.run("cache save 'pkg/*'", assert_error=True)
+    assert (
+        "ERROR: The 'compression.py' plugin does not contain required `tar_extract` or `tar_compress` functions"
+        in c.out
+    )
+
+
+def test_compression_plugin_correctly_load():
+    """Test that the compression plugin is correctly loaded and used on:
+    - cache save/restore
+    - remote upload/download
+    """
+    c = TestClient(default_server_user=True)
+
+    compression_plugin = textwrap.dedent(
+        """
+        import os
+        import tarfile
+        from conan.api.output import ConanOutput
+
+        # xz compression
+        def tar_compress(files, name, dest_dir, compresslevel=None, ref=None, recursive=False, *args, **kwargs):
+            tgz_path = os.path.join(dest_dir, name)
+            ConanOutput(scope=str(ref) if ref else "").info(f"Compressing {name} using compression plugin (xz)")
+            kwargs = {"preset": compresslevel} if compresslevel else {}
+            with tarfile.open(tgz_path, f"w:xz", **kwargs) as tgz:
+                for filename, abs_path in sorted(files.items()):
+                    tgz.add(abs_path, filename, recursive=True)
+            return tgz_path
+
+        def tar_extract(src_path, destination_dir, *args, **kwargs):
```

Comment:
``src_path`` might be ``name``, ``filename`` as variable name, a bit more aligned with the above ``tar_compress(..., name,`` and Python built-in tar module?